src/XmlStringStreamer/Stream/File.php

```diff
@@ -13,27 +13,37 @@ class File implements StreamInterface
     public function __construct($mixed, $chunkSize = 16384, $chunkCallback = null)
     {
         if (is_string($mixed)) {
-            // Treat as filename
-            if (!file_exists($mixed)) {
-                throw new \Exception("File '$mixed' doesn't exist");
+            // Account for common stream/URL wrappers before checking if a file exists
+            $realPath = $mixed;
+            if (preg_match('/^([\w.]+):\/\//', $realPath, $matched)) {
+                if (preg_match('/(http|ftp|php|data|ssh2)/', $matched[1])) {
+                    // Disable file_exists() check
+                    $realPath = null;
+                } else {
+                    // Remove wrapper for file_exists() check.
+                    $realPath = substr($realPath, strlen($matched[0]));
+                }
             }
-            $this->handle = fopen($mixed, "rb");
+            // If there's a real disk path to check, make sure it exists
+            if ($realPath !== null && !file_exists($realPath)) {
+                throw new \Exception("File '$realPath' doesn't exist");
+            }
+            $this->handle = fopen($mixed, 'rb');
             $this->handle;
```

Comment:
While we're here—I noticed this no-op statement and it confused me. Is there a reason for it?

Reply:
Lol yeah no idea. Delete :)